pol-core/pol/decay.cpp

```diff
@@ -259,22 +249,23 @@ void decay_single_thread( void* arg )
           }
           init = false;
         }
-        wx = 0;
-        wy = 0;
+        pos.x( 0 ).y( 0 );
       }
       else
       {
-        if ( ++wx >= gridx )
+        pos += Vec2d( 1, 0 );
```

Comment:
if you consider changing the iterator on the other method, here would also be a good place for it